drivers/include/periph/usbdev.h

```diff
@@ -125,6 +125,20 @@ typedef struct usbdev_ep usbdev_ep_t;
  */
 #define usbdev_ep_buf_t USBDEV_CPU_DMA_REQUIREMENTS uint8_t
 
+/**
+ * @brief USBDEV specific requirement for setting the device address
+ *
+ * The address in the USB device can be set either directly after the SETUP
+ * stage on receipt of the `SET ADDRESS Request` or after the associated status
+ * stage. When the USB device address has to be set depends on the hardware.
+ * If `USBDEV_SET_ADDR_AFTER_STATUS` has the value 1 (default), the address
+ * is only set in the USB device after the status stage. Overwrite it with 0
+ * in `periph_cpu.h` to set the address already directly after the SETUP stage.
+ */
+#ifndef USBDEV_SET_ADDR_AFTER_STATUS
+#define USBDEV_SET_ADDR_AFTER_STATUS    1
```

Comment:
I think you want to include `periph_conf.h` in this file to avoid conflicts when `periph_conf.h` overrides this setting, but is included after this file.

Reply:
Since it is a property of the MCU and not of a board, it would be defined in `periph_cpu.h` if it is defined. The file `periph_cpu.h` is already included. So I wondered if I should call the symbol `USBDEV_CPU_SET_ADDR_AFTER_STATUS` instead.

Reply:
Setting it in `periph_cpu.h` sounds perfect. I don't have a strong opinion on the name, but using the `USBDEV_CPU_` prefix sounds nice as that aligns it with the `USBDEV_CPU_DMA_*` macros. 

`usbdev_synopsys_dwc2.h` (that overrides this) is as far as I can see not included in the stm32 `periph_cpu.h`, so depending on the other includes in the file, it either does or does not override this define for that peripheral.

Reply:
> `usbdev_synopsys_dwc2.h` (that overrides this) is as far as I can see not included in the stm32 `periph_cpu.h`

By intention because of USB cores and the different drivers used. It's included instead in `boards/common/stm32/cfg_usb_otg_*.h`. `usbdev_fs` requires to set the device address after STATUS stage.

The question is how we can ensure that the definition in `usbdev_synopsys_dwc2.h` always comes before the default defintion in `periph/usbdev.h`

Reply:
I changed the name to `USBDEV_CPU_SET_ADDR_AFTER_STATUS`.

Reply:
> The question is how we can ensure that the definition in `usbdev_synopsys_dwc2.h` always comes before the default defintion in `periph/usbdev.h`

That's why I was suggesting to include `periph_conf.h` in `periph/usbdev.h`. `periph_conf.h` must include the `usbdev_synopsys_dwc2.h` if it configures one, pretty much guaranteeing the correct order of includes required here. Other periph headers (`periph/spi.h` and `periph/i2c.h`)  files also include `periph_conf.h`, so I don't see much reason why not to include it also here.